### ISSUE TITLE
error: UNSAFE_componentWillReceiveProps in strict mode is not recommended

### ISSUE THREAD

**Issue by Hrach2003** [ankeetmaini/react-infinite-scroll-component#203]
I am using 
   react 16.13.1
   react-infinite-scroll-component 5.0.5
 
error:   Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memorization techniques or move it to static getDerivedStateFromProps. Learn more at https://fb.me/react-derived-state

Please update the following components: InfiniteScroll

**Comment by ramyNiranjan** [comment]
Hi 
I'm using same verion and get same error massage.

**Comment by carlpadilla** [comment]
Also getting this same error.

**Comment by romadryud** [comment]
Hi also get this error.

**Comment by AsebWebDev** [comment]
+1

**Comment by Imferal** [comment]
I have the same...

**Comment by hartman42** [comment]
Yeah, I get that too... are you guys using it in a function component as well?

**Comment by Imferal** [comment]
> Yeah, I get that too... are you guys using it in a function component as well?

Yes exactly. Do you think it matters?

**Comment by Deivaras** [comment]
Hey guys, is anything done about this?

**Comment by git-hafizh** [comment]
i just get this error, how can i fix this?

**Comment by Deivaras** [comment]
> i just get this error, how can i fix this?

As I said in a message above go with react-infinite-scroller

**Comment by gynekolog** [comment]
There's still open PR https://github.com/ankeetmaini/react-infinite-scroll-component/pull/178 which can solve this issue.
PR is waiting for you @ankeetmaini .

**Comment by WeslleyNasRocha** [comment]
> There's still open PR #178 which can solve this issue.
> PR is waiting for you @ankeetmaini .

there is a conflicting file in the pull request

**Comment by pearlpandz** [comment]
Is this issue fixed? or still open?

**Comment by shweps13** [comment]
+1

**Comment by sanddy-man** [comment]
+1

**Comment by gynekolog** [comment]
New PR #224  without conflicts.
Waiting for @ankeetmaini

**Comment by hodev-dev** [comment]
save problem 
```js{
{
    "react-infinite-scroll-component": "^6.0.0",
}

**Comment by OriAmir** [comment]
Any news relate to that?
I still face that error on version 6.0.0

**Comment by steinroe** [comment]
Same problem here

**Comment by hodev-dev** [comment]
i hope it get fix because its really good library. but project manager wont let push until error get fixed

**Comment by GuiSelair** [comment]
Same problem

**Comment by markoromandic** [comment]
Any news, when can we expect this to be resolved?

**Comment by abdelhamid1A** [comment]
it work with me :)
remove React.StrictMode from index.js
replace:
<React.StrictMode>
    <App />
  </React.StrictMode>,

-----------------------------------------

<>
    <App />
  </>,

**Comment by markoromandic** [comment]
> it work with me :)
> remove React.StrictMode from index.js
> replace:
> 
> ```
> <React.StrictMode>
>        <App />
>   </React.StrictMode>,
> ```
> 
> ```
> <>
>     <App />
>   </>,
> ```

That is for sure not considerable approach...

Check this for more input: https://reactjs.org/docs/strict-mode.html

**Comment by lynda0214** [comment]
Same problem

**Comment by iamdarshshah** [comment]
This warning was tied to the class component lifecycle. v7.1.0 rewrites the component entirely as a function component using `useRef`/`useEffect`, there are no class lifecycle methods left in the codebase. Upgrading to v7.1.0 will resolve this.
